### PR TITLE
SPORTS-195 Add game CRUD

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
- Add CRUD features for GAMES on both back-end and front-end
  - `createGame (name: String, shortName: String)`
  - `editGame (id: Int, name: String, shortName: String)`
  - `deleteGame (id: Int)`
  - `games`

image

### CODE REVIEW

File: api/src/mutations/game.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import * as gameServices from '../services/game';
+
+/**
+ * Mutation for creating a new game.
+ *
+ * @export
+ * @param {*} parent
+ * @param {{ name: string; shortName: string }} { name, shortName }
+ * @param {Context} context
+ * @returns
```
Comment (SAGARACH65) [inline api/src/mutations/game.ts]:
```suggestion
 * @returns {object}
```

File: api/src/mutations/game.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+/**
+ * Mutation to delete an existing game.
+ *
+ * @export
+ * @param {*} parent
+ * @param {{ id: number; }} { id }
+ * @param {Context} context
+ * @returns
```
Comment (SAGARACH65) [inline api/src/mutations/game.ts]:
```suggestion
 * @returns {object}
```

File: api/src/queries/games.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import * as gameServices from '../services/game';
+
+/**
+ * Query for fetching the list of all games.
+ *
+ * @export
+ * @param {*} parent
+ * @param {{ name: string; shortName: string }} { name, shortName }
+ * @param {Context} context
+ * @returns
```
Comment (SAGARACH65) [inline api/src/queries/games.ts]: .

File: api/src/services/game.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import * as HttpStatus from 'http-status-codes';
+import { ApolloError } from 'apollo-server-core';
+
+import Game from '../models/Game';
+
+/**
+ * Create a new game.
+ *
+ * @param payload
+ */
```
Comment (SAGARACH65) [inline api/src/services/game.ts]: let's add a returns block

File: api/src/services/game.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+export async function createGame(payload: { name: string; shortName: string }) {
+  const newGame = await new Game({
+    name: payload.name,
+    shortName: payload.shortName.toLowerCase(),
+    updatedBy: 3
```
Comment (SAGARACH65) [inline api/src/services/game.ts]: Cant we add the user's id who edited it actually. I believe that data is received from the middleware we added for verifying tokens

File: api/src/services/game.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  return updatedGame.serialize();
+}
+
+/**
+ * Delete an existing game.
+ *
+ * @export
+ * @param {number} id
+ * @returns
```
Comment (SAGARACH65) [inline api/src/services/game.ts]:
```suggestion
 * @returns {{message:string}}
```

File: admin-app/src/enums/modals.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import AddEditGameFormModal from '../components/games/AddEditGameFormModal.vue';
```
Comment (SAGARACH65) [inline admin-app/src/enums/modals.ts]: Why are we importing a component as an modal inside enums?

File: api/src/services/game.ts (hunk 5 above)
Comment (sanjeevkpandit) [reply api/src/services/game.ts]: Yeah, right :smile:

**Pull request merged**